pkg/test/case.go

```diff
@@ -397,6 +397,14 @@ func (t *Case) determineNamespace() *namespace {
 	if t.PreferredNamespace == "" {
 		ns.Name = fmt.Sprintf("kuttl-test-%s", petname.Generate(2, "-"))
 		ns.AutoCreated = true
+	} else {
+		exist, err := t.NamespaceExists(t.PreferredNamespace)
+		if err != nil {
+			t.Logger.Logf("found namespace %s err, %s", t.PreferredNamespace, err.Error())
```

Comment:
I don't think we want to log+ignore an error when retrieving namespace. If we're not able to even look up its existence then creating it is even less likely to succeed. How about changing `determineNamespace` to return `(*namespace, error)` and then failing the run if `err != nil` in its caller(s)?

Reply:
yes, I aggree. This looks much better.